packages/esm-patient-test-results-app/src/filter/filter-reducer.ts

```diff
@@ -1,31 +1,39 @@
-import { ReducerAction, ReducerState, ReducerActionType } from './filter-types';
+import { ReducerAction, ReducerState, ReducerActionType, TreeParents, TreeNode } from './filter-types';
 
 export const getName = (prefix, name) => {
   return prefix ? `${prefix}-${name}` : name;
 };
 
-const computeParents = (prefix, node) => {
-  var parents = {};
-  var leaves = [];
-  var tests = [];
-  var lowestParents = [];
+const computeParents = (
+  prefix: string,
+  node: TreeNode,
+): {
+  parents: TreeParents;
+  leaves: Array<string>;
+  tests: Array<[string, TreeNode]>;
+  lowestParents: Array<TreeNode>;
+} => {
+  let parents: TreeParents = {};
+  const leaves: Array<string> = [];
+  const tests: Array<[string, TreeNode]> = [];
+  const lowestParents: Array<TreeNode> = [];
   if (node?.subSets?.length && node.subSets[0].obs) {
     // lowest parent
-    let activeLeaves = [];
+    const activeLeaves: Array<string> = [];
     node.subSets.forEach((leaf) => {
       if (leaf.hasData) {
         activeLeaves.push(leaf.flatName);
       }
     });
-    let activeTests = [];
+    const activeTests: Array<[string, TreeNode]> = [];
     node.subSets.forEach((leaf) => {
-      if (leaf.obs.length) {
+      if (Array.isArray(leaf?.obs) && leaf.obs.length > 0) {
         activeTests.push([leaf.flatName, leaf]);
       }
     });
     leaves.push(...activeLeaves);
     tests.push(...activeTests);
-    lowestParents.push({ flatName: node.flatName, display: node.display });
+    lowestParents.push({ flatName: node.flatName, display: node.display, ...node });
```

Comment:
Is this change where you're spreading extra object properties from the `node` into `lowestParents` intentional?

Reply:
Made a new Type `LowestNode` and removed `...node`